homeassistant/components/livebox/binary_sensor.py

```diff
@@ -0,0 +1,79 @@
+"""Livebox binary sensor entities."""
+from homeassistant.util import Throttle
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
+    DEVICE_CLASS_CONNECTIVITY,
+)
+
+from . import DOMAIN, SCAN_INTERVAL
+from .const import TEMPLATE_SENSOR
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer binary sensor setup to the shared sensor module."""
+    box = hass.data[DOMAIN]
+    async_add_entities([InfoSensor(box, config_entry)], True)
+
+
+class InfoSensor(BinarySensorDevice):
+    """Representation of a livebox sensor."""
+
+    device_class = DEVICE_CLASS_CONNECTIVITY
+
+    def __init__(self, box, config_entry):
+        """Initialize the sensor."""
+
+        self._box = box
+        self._box_id = config_entry.data["box_id"]
+        self._state = None
+        self._datas = None
```

Comment:
This doesn't have to be an instance attribute. We just use it as a local variable.